rep-2002.rst

```diff
@@ -0,0 +1,130 @@
+
+
+Support status
+==============
+
+The rolling release itself will not be considered a supported platform.
+Distribution syncs will be performed on a regular cadence just like stable distributions but regressions may not block a rolling release sync.
+During the pre-release phase for stable ROS distributions releases in the rolling distribution may be blocked or held back in order to conform to the support requirements of the upcoming stable release.
+
+
+Release process
+===============
+
+Releasing stable ROS distributions from a rolling release
+---------------------------------------------------------
+
+When a new stable release is being prepared, the rolling release will be used as a staging ground to prepare that release.
+Updates and new package releases made to the rolling release will be propagated automatically to the stable release.
+At an announced time prior to the official stable release, the stable distribution will be created and packages with successful builds on the rolling release will be automatically released via bloom into the stable release.
+Once the stable distribution has been created and the automatic bloom process has been run, changes to the rolling release may not be automatically propagated to the stable release although 
+
+
+Updating the rolling release platforms
+--------------------------------------
+
+The rolling release is intended to remain availabe indefinitely and so will not be retired or reach end-of-support status without an update to this REP and the ROS distribution release process.
+However, this does mean that platform changes during the life of the rolling distribution are going to occur and the rolling release may target platforms that are themselves unstable or in a pre-release state.
+When a platform change is coming, the intended date of change will be announced ahead of time.
+Once the platforms have been updated, everything currently in the rolling release will be automatically re-released via bloom to target the updated release platforms.
+
+
+Bootstrapping the rolling release
+---------------------------------
+
+The first release of the rolling rosdistro will use the same procedure we've used for all stable rosdistros previously.
+Packages will be released via bloom into an empty distribution until the distribution is relatively complete.
+Package maintainers may then choose to make releases into the rolling rosdistro.
+
+
+Recommendations for package maintainers
+=======================================
+
+Versioning between rolling and stable distributions
+---------------------------------------------------
+
+When a new stable release is created, the ROS core team strongly recommends that the next release made into the rolling release bump at least the minor version number.
```

Comment:
It's possible a maintainer could get into a situation where they are unable to release their package in time for a new stable release and so the version of their package that happened to be in the rolling release gets released instead. Then they are faced with the options of releasing breaking changes into the stable release or waiting a whole release cycle. It might be nice to have the option to opt-out of upcoming stable releases, so that it gives maintainers a chance to release into a stable distro a bit later than the initial release.

Reply:
The proposed process is the same as for Debian. If a maintainer missed the deadline the previously available version is being used - and that is being done since it is preferred to include that rather than nothing.

If we would allow a maintainer to opt-out that implies that potentially a lot of downstream packages are being prevented from being released. I would rather not allow that.

Reply:
> If we would allow a maintainer to opt-out that implies that potentially a lot of downstream packages are being prevented from being released.

I see your point. I presume maintainers will still have the option to opt-in to the rolling release.

Reply:
> I presume maintainers will still have the option to opt-in to the rolling release.

That's a good point. As a maintainer, I could wait until X-turtle is released, and then release my package directly there. Since the rolling distro has already moved on, my package would not be included in the rolling release unless I explicitly bloom it for the rolling release? Which means I'd need to manually bloom it again for Y-turtle. Is that the case @nuclearsandwich ?

https://github.com/ros-infrastructure/rep/blob/1e7acd97e522658b528024efa7f560b91953e405/rep-2002.rst#L41

If that's the case, then the sentence above also goes the other way, i.e.: _and changes to the stable release may not be propagated to the rolling release._

Reply:
>  Which means I'd need to manually bloom it again for Y-turtle. Is that the case @nuclearsandwich ?

Yes. Once a stable release has been made, changes there will not affect the rolling release.